RNA-seq/05-nb_cell_line_DESeq2.Rmd

```diff
@@ -207,6 +207,50 @@ summary(results(deseq_object), alpha = 0.05)
 readr::write_tsv(deseq_df, path = deseq_df_file)
 ```
 
+#### Visualize differential expression results with a heatmap
+
+Let's get an overall picture of our results using a heatmap. 
+Before we can do this, we will need to transform our data so that when we make
+the heatmap the genes are on a more similar scale and the heatmap is 
+interpretable. 
+We will use variance stablizing transformation (VST) from DESeq2 to do this.
+Refer to the help page or the DESeq2 docs for more information. 
+
+```{r}
+vst_data <- vst(ddset)
+```
+
+Need to extract the normalized gene matrix from the `vst_data` object. 
+
+```{r}
+vst_data <- assay(vst_data)
+```
+
+We will make a heatmap of only the significant genes, so let's obtain the gene 
+IDs that have an adjusted p-value less than 0.05.
+
+```{r}
+signif_genes <- deseq_df %>%
+  dplyr::filter(padj < 0.05)
+```
+
+Now we will use the information from `signif_genes` to filter `vst_data` to 
+only have significant genes' data. 
+
+```{r}
+signif_vst_data <- vst_data %>%
+  as.data.frame() %>%
+  dplyr::filter(rownames(vst_data) %in% signif_genes$Gene)
+```
+
+Let's make the results into a heatmap using the `pheatmap` package. 
+
+```{r}
+pheatmap::pheatmap(signif_vst_data, show_rownames = FALSE)
```

Comment:
I think
```R
pheatmap::pheatmap(signif_vst_data, 
                   show_rownames = FALSE,
                   scale = "row",
                   colorRampPalette(c("deepskyblue",
                                      "black",
                                      "yellow"))(25))
```
would be closer to what folks are expecting to see. You should make a note, then, about the consequences/interpretation of scaling by row. There is a scale by row step in the scRNA-seq exercise notebook, do you cover that anywhere else? We talk a bit about z-scoring genes in the ML section but that wouldn't be until after the scRNA-seq notebook.

Reply:
No, I don't think we cover that anywhere else. I can add a sentence or two about it here.